marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -357,3 +368,21 @@ mkIndexerStream coordinator = S.foldM_ step initial finish
 
     finish :: Coordinator -> IO ()
     finish _ = pure ()
+
+runIndexers
+  :: FilePath -> Shelley.NetworkId -> [ChainPoint]
+  -> (S.Stream (S.Of (ChainSyncEvent (BlockInMode CardanoMode))) IO r -> IO ())
```

Comment:
Like said above, we can combine the `startIndexer` with this function. Once we do that, we don't need to provide the `Stream` because we'll have access to the `Coordinator` and everything can be created in one go.

Reply:
A lot better now, a42e0cfe8